src/main/java/org/primefaces/model/datepicker/DefaultDateMetadata.java

```diff
@@ -62,12 +75,15 @@ public boolean equals(Object obj) {
             return false;
         }
         final DefaultDateMetadata other = (DefaultDateMetadata) obj;
-        return this.disabled == other.disabled;
+        if (this.disabled != other.disabled) {
+            return false;
+        }
+        return Objects.equals(this.styleClass, other.styleClass);
     }
```

Comment:
Why not just make your equals and hashcode methods...
```java
@Override
    public int hashCode() {
        return Objects.hash(disabled, styleClass);
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (!(obj instanceof DefaultDateMetadata)) {
            return false;
        }
        DefaultDateMetadata other = (DefaultDateMetadata) obj;
        return disabled == other.disabled && Objects.equals(styleClass, other.styleClass);
    }
```

Reply:
I did not bother writing them. It's how they were generated by NetBeans.

Reply:
Ahh mine above were generated by Eclipse.  Interesting!

Reply:
.. doesn't matter, I'll change them

Reply:
OK cool yeah its juts that `Objects.hash()` already does all the magic inside of it so its strange Netbeans generates like that.